akka-actor-typed/src/main/scala/akka/actor/typed/receptionist/Receptionist.scala

```diff
@@ -269,15 +281,65 @@ object Receptionist extends ExtensionId[Receptionist] {
     def isForKey(key: ServiceKey[_]): Boolean
 
     /**
-     * Scala API
+     * Scala API: Return the reachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be all registered instances
+     * for a service key.
+     *
+     * For a clustered `ActorSystem` it only contain services on nodes that
+     * are not seen as unreachable (note that they could have still have become
+     * unreachable between this message being sent and the receiving actor processing it).
+     *
+     * For a list including both reachable and unreachable instances see [[#allServiceInstances]]
      *
      * Also, see [[ServiceKey.Listing]] for more convenient pattern matching
      */
     def serviceInstances[T](key: ServiceKey[T]): Set[ActorRef[T]]
 
-    /** Java API */
+    /**
+     * Java API: Return the reachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be all registered instances
+     * for a service key.
+     *
+     * For a clustered `ActorSystem` it only contain services on nodes that has
+     * are not seen as unreachable (note that they could have still have become
+     * unreachable between this message being sent and the receiving actor processing it).
+     *
+     * For a list including both reachable and unreachable instances see [[#getAllServiceInstances]]
+     */
     def getServiceInstances[T](key: ServiceKey[T]): java.util.Set[ActorRef[T]]
 
+    /**
+     * Scala API: Return both the reachable and the unreachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be the same as [[#serviceInstances]].
+     *
+     * For a clustered `ActorSystem` this include both services on nodes that are reachable
+     * and nodes that are unreachable.
+     */
+    def allServiceInstances[T](key: ServiceKey[T]): Set[ActorRef[T]]
+
+    /**
+     * Java API: Return both the reachable and the unreachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be the same as [[#getServiceInstances]].
+     *
+     * For a clustered `ActorSystem` this include both services on nodes that are reachable
+     * and nodes that are unreachable.
+     */
+    def getAllServiceInstances[T](key: ServiceKey[T]): java.util.Set[ActorRef[T]]
+
+    /**
+     * Returns `true` if this listing is only about reachability changes and `false` if new services were
+     * actually added or removed to the receptionist. Useful for subscribers only concerned with [[#allServiceInstances]]
+     * or [[#getAllServiceInstances]] that can then ignore `Listing`s related to reachability.
+     *
+     * In a non-clustered `ActorSystem` this will be `false` for all listings.
+     * For `Find` queries and the initial listing for a `Subscribe` this will always be `false`.
+     */
+    def onlyReachabilityChanged: Boolean
```

Comment:
I'm not sure this is a good name change. Nowhere else in the API signatures we have reachability. Could be good to not tie it too close to unreachable/reachable so that we can use it for preferred service instances based on other other health or load metrics in the future.

```
def isAllServiceInstancesChanged: Boolean
```

Reply:
I think `isAllServiceInstancesChanged` is hard to parse though. What about `wasRegisteredActorsChanged`?

Reply:
`wasServicesAddedOrRemoved`

Reply:
that is a good name!